tests/common/test_additional_secrets_redaction.py

```diff
@@ -0,0 +1,345 @@
+"""Test customizing security admin segment traits."""
+
+import unittest
+from unittest.mock import Mock, patch
+
+import __init__
+
+import tests.common.test_common_constants as TestCommonConstants
+import tests.data_set.test_data_set_constants as TestDataSetConstants
+import tests.group.test_group_constants as TestGroupConstants
+import tests.resource.test_resource_constants as TestResourceConstants
+import tests.user.test_user_constants as TestUserConstants
+from pyracf import (
+    DataSetAdmin,
+    GroupAdmin,
+    ResourceAdmin,
+    SecurityRequestError,
+    UserAdmin,
+)
+
+# Resolves F401
+__init__
+
+
+class TestAdditionalSecretsRedaction(unittest.TestCase):
+    maxDiff = None
+
+    # ============================================================================
+    # User Administration
+    # ============================================================================
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_user_admin_custom_secret_redacted_on_success(
+        self,
+        call_racf_mock: Mock,
+    ):
+        user_admin = UserAdmin(additional_secret_traits=["omvs:uid"])
+        call_racf_mock.side_effect = [
+            TestUserConstants.TEST_EXTRACT_USER_RESULT_BASE_ONLY_SUCCESS_XML,
+            TestUserConstants.TEST_ALTER_USER_RESULT_EXTENDED_SUCCESS_XML,
+        ]
+        result = user_admin.alter(
+            "squidwrd",
+            traits=TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_EXTENDED,
+        )
+        self.assertEqual(
+            result,
+            TestCommonConstants.TEST_ALTER_USER_RESULT_SUCCESS_UID_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_EXTENDED["omvs:uid"],
+            result,
```

Comment:
Not sure if these `assertNotIn` assertions actually work correctly since result is a dictionary. You might need to stringify the result dictionary in order for this to work.

Reply:
That is an excellent point and has now been resolved.